mesonbuild/interpreter/interpreter.py

```diff
@@ -1067,14 +1068,22 @@ def func_project(self, node, args, kwargs):
 
     @FeatureNewKwargs('add_languages', '0.54.0', ['native'])
     @permittedKwargs({'required', 'native'})
-    @stringArgs
-    def func_add_languages(self, node, args, kwargs):
+    @typed_pos_args('add_language', varargs=str)
+    @typed_kwargs(
+        'add_language',
+        type_checking.REQUIRED,
```

Comment:
This actually might not even work, because we'd end up with circular dependencies where the `common_kwargs` requires `intpreterobjects` but `interpreterobjects` requires `common_kwargs`. I guess we can punt on that for the moment while we land the actually implementation.

Reply:
That's out of scope for this PR now, let's see what we can do later.